src/coreclr/vm/prestub.cpp

```diff
@@ -1059,7 +1045,437 @@ PCODE MethodDesc::JitCompileCodeLocked(PrepareCodeConfig* pConfig, JitListLockEn
+        }
+
+        // Return from the generated stub
+        pCode->EmitRET();
+
+        {
+            UINT maxStack;
+            size_t cbCode = sl.Link(&maxStack);
+            DWORD cbSig = sl.GetLocalSigSize();
+
+            COR_ILMETHOD_DECODER* pILHeader = ilResolver->AllocGeneratedIL(cbCode, cbSig, maxStack);
+            BYTE* pbBuffer = (BYTE*)pILHeader->Code;
+            BYTE* pbLocalSig = (BYTE*)pILHeader->LocalVarSig;
+            _ASSERTE(cbSig == pILHeader->cbLocalVarSig);
+            sl.GenerateCode(pbBuffer, cbCode);
+            sl.GetLocalSig(pbLocalSig, cbSig);
+
+            // Store the token lookup map
+            ilResolver->SetTokenLookupMap(sl.GetTokenLookupMap());
+            ilResolver->SetJitFlags(CORJIT_FLAGS(CORJIT_FLAGS::CORJIT_FLAG_IL_STUB));
+
+            *resolver = (DynamicResolver*)ilResolver;
+            *methodILDecoder = pILHeader;
+        }
+
+        ilResolver.SuppressRelease();
+    }
+}
+
+bool MethodDesc::TryGenerateUnsafeAccessor(DynamicResolver** resolver, COR_ILMETHOD_DECODER** methodILDecoder)
+{
+    STANDARD_VM_CONTRACT;
+    _ASSERTE(resolver != NULL);
+    _ASSERTE(methodILDecoder != NULL);
+    _ASSERTE(*resolver == NULL && *methodILDecoder == NULL);
+    _ASSERTE(IsIL());
+    _ASSERTE(GetRVA() == 0);
+
+    // The UnsafeAccessorAttribute is applied to methods with an
+    // RVA of 0 (for example, C#'s extern keyword).
+    const void* data;
+    ULONG dataLen;
+    HRESULT hr = GetCustomAttribute(WellKnownAttribute::UnsafeAccessorAttribute, &data, &dataLen);
+    if (hr != S_OK)
+        return false;
 
+    UnsafeAccessorKind kind;
+    SString name;
+
+    CustomAttributeParser ca(data, dataLen);
+    if (!TryParseUnsafeAccessorAttribute(this, ca, kind, name))
+        ThrowHR(COR_E_BADIMAGEFORMAT, BFA_INVALID_UNSAFEACCESSOR);
+
+    GenerationContext context{ kind, this };
+
+    // Parse the signature to determine the type to use:
+    //  * Constructor access - examine the return type
+    //  * Instance member access - examine type of first parameter
+    //  * Static member access - examine type of first parameter
+    TypeHandle retType;
+    TypeHandle firstArgType;
+    retType = context.DeclarationSig.GetRetTypeHandleThrowing();
+    UINT argCount = context.DeclarationSig.NumFixedArgs();
+    if (argCount > 0)
+    {
+        context.DeclarationSig.NextArg();
+        firstArgType = context.DeclarationSig.GetLastTypeHandleThrowing();
+    }
+
+    // Using the kind type, perform the following:
+    //  1) Validate the basic type information from the signature.
+    //  2) Resolve the name to the appropriate member.
+    switch (context.Kind)
+    {
+    case UnsafeAccessorKind::Constructor:
+        // A return type is required for a constructor, otherwise
+        // we don't know the type to construct.
+        // The name is defined by the runtime and should be empty.
+        if (context.DeclarationSig.IsReturnTypeVoid() || retType.IsNull() || !name.IsEmpty())
```

Comment:
```suggestion
        if (context.DeclarationSig.IsReturnTypeVoid() ||  !name.IsEmpty())
```
retType cannot be null here. `GetRetTypeHandleThrowing()` above should always throw instead of returning null.